CONTRIBUTING.md

```diff
@@ -42,19 +42,35 @@ our [Style Guide](docs/StyleGuide.md).
 
     * Run the [test suite](#testing) and make sure everything passes.
 
-    * When committing, be sure to write good commit messages. Stylistically,
+    * When committing, be sure to write good commit messages according to [these
+      seven rules](https://chris.beams.io/posts/git-commit/#seven-rules). Doing 
+      `git commit` prints a message if any of the rules is violated. 
+      Stylistically,
       we use commit message titles in the imperative tense, e.g., `Add
       merge-append query optimization for time aggregate`.  In the case of
       non-trivial changes, include a longer description in the commit message
       body explaining and detailing the changes.  That is, a commit message
       should have a short title, followed by a empty line, and then
       followed by the longer description.
 
+    * When committing, add which GitHub issue is fixed or closed by the commit. 
+      For example, if the commit fixes bug 123, add a line at the end of the 
+      commit message with  `Fixes #123`, if the commit implements feature 
+      request 321, add a line at the end of the commit message `Closes #321`.
+      This will be recognized by GitHub. It will close the corresponding issue 
+      and place a hyperlink under the number.
```

Comment:
Is it a requirement that we should use "Fixes" for bugs and "Closes" for features?

Reply:
What about if it fixes a private issue? Should we still have "Fixes" or "Closes"?

Reply:
> Is it a requirement that we should use "Fixes" for bugs and "Closes" for features?

It is not necessary to use exactly these keywords. More of them available for the use. Thus I mention the keywords in the example.
It might be confusing, but I haven't found better formulation, which will be correct and short.

Reply:
> What about if it fixes a private issue? Should we still have "Fixes" or "Closes"?

This contributing document is for everyone who contributes to TimescaleDB repo, including external and internal contributors. External contributors don't have access and are not aware about private repos.

Reply:
@mkindahl I committed an improvement, which helps to address your comments. It is in a separate commit for now. Does it improve? If so, I will squash the commits.

Reply:
> > What about if it fixes a private issue? Should we still have "Fixes" or "Closes"?
> 
> This contributing document is for everyone who contributes to TimescaleDB repo, including external and internal contributors. External contributors don't have access and are not aware about private repos.

Well, the same applies to referencing other repos in general. I think the real question is if you should ever refer to repositories outside timescaledb in any commit message. I also think the answer is "No". If there are discussions around this, it should be in the associated issue and in some cases in the PR discussion, but not the commit messages.

Reply:
Private repos are not visible for external users, so issues from private repos should not be mentioned. 

I don't think it is possible to close issues in another repo, so it only works to mention them.